### PR TITLE
Allow underscores in the hostmask

### PR DESCRIPTION
**Pull request by brianjmurrell**
While the "rules" may say that _ is not allowed in host and/or
domainnames, the real worl reality is that they are widely used
and not breaking anything.  So embracing the robustness principle
and accepting the world as it is rather than how it should be, allow
the use of _ in host/domainnames.

Fixes: #188

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Can anyone please review this PR?

**Comment by brianjmurrell** [comment]
What is holding up this of from being landed?

File: app/src/main/java/org/connectbot/transport/SSH.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public SSH(HostBean host, TerminalBridge bridge, TerminalManager manager) {
 	private final static int AUTH_TRIES = 20;
 
 	private static final Pattern hostmask = Pattern.compile(
-			"^(.+)@(([0-9a-z.-]+)|(\\[[a-f:0-9]+\\]))(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
+			"^(.+)@(([0-9a-z.-_]+)|(\\[[a-f:0-9]+\\]))(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
```
Comment (kruton) [inline app/src/main/java/org/connectbot/transport/SSH.java:111]: The hyphen must be the last in the regex or it changes the meaning to "all characters starting with . (0x2e) and extending to _ (0x5f)".

File: app/src/main/java/org/connectbot/transport/Telnet.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ public class Telnet extends AbsTransport {
 
 	static final Pattern hostmask;
 	static {
-		hostmask = Pattern.compile("^([0-9a-z.-]+)(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
+		hostmask = Pattern.compile("^([0-9a-z.-_]+)(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
```
Comment (kruton) [inline app/src/main/java/org/connectbot/transport/Telnet.java:71]: The hyphen must be the last in the regex or it changes the meaning to "all characters starting with . (0x2e) and extending to _ (0x5f)".

**Review by kruton — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/connectbot/transport/SSH.java (hunk 1 above)
Comment (brianjmurrell) [reply app/src/main/java/org/connectbot/transport/SSH.java:111]: Done.

File: app/src/main/java/org/connectbot/transport/Telnet.java (hunk 2 above)
Comment (brianjmurrell) [reply app/src/main/java/org/connectbot/transport/Telnet.java:71]: Done.

**Review by kruton — APPROVED** [review]

**Pull request merged**